engine/engine.py

```diff
@@ -0,0 +1,159 @@
+"""
```

Comment:
This is part of refactoring the modelmaker. We now have an `engine` module which steers the training/testing/evaluation procedure independent of the model creation.